gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/shared_policy_group/use_case/DeleteSharedPolicyGroupUseCase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.shared_policy_group.use_case;
+
+import io.gravitee.apim.core.DomainService;
+import io.gravitee.apim.core.audit.domain_service.AuditDomainService;
+import io.gravitee.apim.core.audit.model.AuditInfo;
+import io.gravitee.apim.core.audit.model.AuditProperties;
+import io.gravitee.apim.core.audit.model.EnvironmentAuditLogEntity;
+import io.gravitee.apim.core.shared_policy_group.crud_service.SharedPolicyGroupCrudService;
+import io.gravitee.apim.core.shared_policy_group.model.SharedPolicyGroup;
+import io.gravitee.apim.core.shared_policy_group.model.SharedPolicyGroupAuditEvent;
+import io.gravitee.common.utils.TimeProvider;
+import java.util.Map;
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@DomainService
+public class DeleteSharedPolicyGroupUseCase {
+
+    private final SharedPolicyGroupCrudService sharedPolicyGroupCrudService;
+    private final AuditDomainService auditService;
+
+    public Output execute(Input input) {
+        var sharedPolicyGroupToDelete =
+            this.sharedPolicyGroupCrudService.getByEnvironmentId(input.auditInfo().environmentId(), input.sharedPolicyGroupId());
+
+        this.sharedPolicyGroupCrudService.delete(sharedPolicyGroupToDelete.getId());
```

Comment:
note for future us: we will probably need to check if the spg is DEPLOYED, and undeploy it first (or make the call fail ? )

Reply:
yes. I'm not sure how to do the fail if the undeploy fail 🤔